### ISSUE TITLE
Checkout fails on Windows

### ISSUE THREAD

**Issue by tjorim** [home-assistant/devices-experiment#7]
Getting this error when cloning the repo on Windows:

```
Cloning into 'C:\Users\tiele\devices-experiment'...
remote: Enumerating objects: 3288, done.        
remote: Counting objects: 100% (3288/3288), done.        
remote: Compressing objects: 100% (719/719), done.        
remote: Total 3288 (delta 2227), reused 3128 (delta 2076), pack-reused 0        
Receiving objects: 100% (3288/3288), 218.89 KiB | 5.21 MiB/s, done.
Resolving deltas: 100% (2227/2227), done.
error: invalid path 'devices/august/August Home Inc./AK-R1/commission.md'
fatal: unable to checkout working tree
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/'
```

**Comment by tjorim** [comment]
I could circumvent this issue with `git config --global core.protectNTFS false` but now getting another one:

```
Cloning into 'C:\Users\tiele\devices-experiment'...
remote: Enumerating objects: 3288, done.        
remote: Counting objects: 100% (3288/3288), done.        
remote: Compressing objects: 100% (719/719), done.        
remote: Total 3288 (delta 2227), reused 3128 (delta 2076), pack-reused 0        
Receiving objects: 100% (3288/3288), 218.89 KiB | 4.97 MiB/s, done.
Resolving deltas: 100% (2227/2227), done.
fatal: cannot create directory at 'devices/enphase_envoy/Enphase/Envoy, phases: 2, phase mode: split, net-consumption CT, production CT': Invalid argument
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/'
```

Edit: list of incompatible files and directories (checkout ok after deleting them)
- `devices/sonos/Sonos/Play:1`
- `devices/sonos/Sonos/Connect:Amp`
- `devices/rainmachine/RainMachine/Version 3 (API: 4.6.1)`
- `devices/enphase_envoy/Enphase/Envoy, phases: 2, phase mode: split, net-consumption CT, production CT`
- `devices/synology_dsm/WDC     /WD100EFAX-68LHPN0`: No such file or directory

**Comment by tjorim** [comment]
Setting protectNTFS to false is not great, it now renamed all folders ending with a `.` to remove the dot.

**Comment by balloob** [comment]
Fixed in 8e4ad6ccc54162d60dca2bcd42e375cf93964ff5
